modules/document_repository/jsx/docIndex.js

```diff
@@ -0,0 +1,214 @@
+import {Tabs, TabPane} from 'Tabs';
+import DocUploadForm from './uploadForm';
+import DocCategoryForm from './categoryForm';
+import Loader from 'Loader';
+import FilterableDataTable from 'FilterableDataTable';
+
+class DocIndex extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoaded: false,
+      data: {},
+      error: false,
+      newCategory: false,
+    };
+
+    /**
+     * Set filter to the element's ref for filtering
+     */
+    this.filter = null;
+    this.setFilterRef = (element) => {
+      this.filter = element;
+    };
+
+    // Bind component instance to custom methods
+    this.fetchData = this.fetchData.bind(this);
+    this.formatColumn = this.formatColumn.bind(this);
+    this.newCategoryState = this.newCategoryState.bind(this);
+  }
+   newCategoryState() {
+        this.setState({newCategory: true});
+    }
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
```

Comment:
`componentDidMount` should be the first function in the class.

Reply:
is there a reason for that?